src/gevent/pool.py

```diff
@@ -714,19 +717,39 @@ def free_count(self):
             return 1
         return max(0, self.size - len(self))
 
-    def add(self, greenlet):
+    def add(self, greenlet, blocking=True, timeout=None):
         """
-        Begin tracking the given greenlet, blocking until space is available.
+        Begin tracking the given greenlet, possibly blocking until space is
+        available.
+
+        :keyword bool blocking: If True (the default), this function will block
+        until the pool has space or a timeout occurs.
+        :keyword float timeout: The maximum number of seconds this method will block.
+
+        :return: True if the greenlet was added; False if a Timeout occured or
+        if blocking is True and the pool is full.
```

Comment:
The docstring needs `.. versionchanged::` explaining the changes.

Reply:
Wasn't sure whether this would make the cut for 1.3.0 so I omitted it all together. :) Will add.